index.js

```diff
@@ -292,31 +280,36 @@ class HydraExpress {
   * @private
   * @return {undefined}
   */
-  start(resolve, _reject) {
+  start(resolve, reject) {
     let serviceInfo;
     return hydra.init(this.config, this.testMode)
-      .then((config) => {
-        this.config = config;
-        return Promise.series(this.registeredPlugins, (plugin) => plugin.setConfig(config));
-      })
-      .catch((err) => {
-        this.log('error', {err});
-        process.exit(1);
-      })
-      .then(() => hydra.registerService())
-      .then((_serviceInfo) => {
-        serviceInfo = _serviceInfo;
-        this.log('start', `${hydra.getServiceName()} (v.${hydra.getInstanceVersion()}) server listening on port ${this.config.hydra.servicePort}`);
-        this.log('info', `Using environment: ${this.config.environment}`);
-        this.initService();
-        return Promise.series(this.registeredPlugins, (plugin) => plugin.onServiceReady());
-      })
-      .then(() => Promise.delay(2000))
-      .then(() => resolve(serviceInfo))
-      .catch((err) => {
-        this.log('error', {err});
-        process.emit('cleanup');
+    .then((config) => {
+      this.config = config;
+      let proms = [];
+      this.registeredPlugins.forEach(plugin => {
+        proms.push(plugin.setConfig(config));
+      });
+      return Q.all(proms);
+    })
+    .then(() => hydra.registerService())
+    .then((_serviceInfo) => {
+      serviceInfo = _serviceInfo;
+      this.initService();
+      let proms = [];
+      this.registeredPlugins.forEach(plugin => {
+        proms.push(plugin.onServiceReady());
       });
+      return Q.all(proms);
+    })
+    .then(() => {
+      resolve(serviceInfo);
+      this.ready.resolve();
+    })
+    .catch((err) => {
+      this.ready.reject(err);
+      process.emit('cleanup');
+      reject(err);
+    });
   }
```

Comment:
Removes logs and startup delay. Resolves the ready promise.
This is the exact code we're already using (via utils-hydra overwrite) but without lodash.